src/views/forms/FormAccountRestrictionTransaction/FormAccountRestrictionTransactionTs.ts

```diff
@@ -0,0 +1,262 @@
+
+    @Prop({ default: null })
+    readonly restrictionToBeDeleted: any;
+
+    /**
+     * Form items
+     * @var {any}
+     */
+    public formItems = {
+        signerAddress: '',
+        recipientRaw: '',
+        mosaicIdRaw: '',
+        transactionType: null,
+        direction: RestrictionDirection.INCOMING,
+        blockType: RestrictionBlockType.BLOCK,
+        maxFee: 0,
+    };
+
+    protected isDeleteMode = false;
+
+    public accountAddressRestrictions: AccountRestriction[];
+    public accountMosaicRestrictions: AccountRestriction[];
+    public accountOperationRestrictions: AccountRestriction[];
+
+    /**
+     * Reset the form with properties
+     * @return {void}
+     */
+    protected resetForm() {
+        this.formItems.maxFee = this.defaultFee;
+        this.formItems.signerAddress = this.currentSignerAddress.plain();
+
+        if (this.restrictionToBeDeleted) {
+            this.isDeleteMode = true;
+            if (this.restrictionTxType === AccountRestrictionTxType.ADDRESS) {
+                this.formItems.recipientRaw = this.restrictionToBeDeleted.value;
+            } else if (this.restrictionTxType === AccountRestrictionTxType.MOSAIC) {
+                this.formItems.mosaicIdRaw = this.restrictionToBeDeleted.value;
+            } else {
+                const restriction: AccountRestriction = this.restrictionToBeDeleted.hiddenData.rowObject;
+                this.formItems.transactionType = restriction.values[0];
+            }
+            this.formItems.direction = this.restrictionToBeDeleted.direction;
+            this.formItems.blockType = this.restrictionToBeDeleted.action;
+        } else {
+            this.formItems.recipientRaw = '';
+            this.formItems.mosaicIdRaw = '';
+            this.formItems.transactionType = null;
+            this.formItems.direction =
+                this.restrictionTxType === AccountRestrictionTxType.TRANSACTION_TYPE
+                    ? RestrictionDirection.OUTGOING
+                    : RestrictionDirection.INCOMING;
+        }
+    }
+
+    /**
+     * Recipient used in the transaction
+     * @readonly
+     * @protected
+     * @type {(Address | NamespaceId)}
+     */
+    protected get instantiatedRecipient(): Address | NamespaceId {
+        const { recipientRaw } = this.formItems;
+        if (AddressValidator.validate(recipientRaw)) {
+            return Address.createFromRawAddress(recipientRaw);
+        } else if (AliasValidator.validate(recipientRaw)) {
+            return new NamespaceId(recipientRaw);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Getter for transactions that will be staged
+     * @throws {Error} If not overloaded in derivate component
+     */
+    protected getTransactions(): Transaction[] {
+        return [this.createTransaction(this.restrictionTxType)];
+    }
+
+    private createTransaction(restrictionTxType: AccountRestrictionTxType): Transaction {
```

Comment:
I think here we should add some checks for better ux. 
- MosaicAddress restriction need a mosaicGlobalRestriction pre-exits
- Operation restriction can only be added for outgoing transaction types.

Reply:
> MosaicAddress restriction need a mosaicGlobalRestriction pre-exits
What is MosaicAddress? account mosaic, address restrictions maybe?

>Operation restriction can only be added for outgoing transaction types.
Is there any programmatic way to achieve this filtering? If not, where can I find the list?